src/deepEqual.js

```diff
@@ -20,7 +20,32 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  */
 function deepEqual(a, b) {
-  // write code here
+  if (typeof a === 'object' && typeof b === 'object') {
+    if (a === null || b === null) {
+      return a === b;
+    } else {
+      return areObjectsEqual(a, b);
+    }
+  } else {
+    return a === b;
+  }
+}
+
+function areObjectsEqual(obj1, obj2) {
+  if (Object.keys(obj1).length !== Object.keys(obj2).length) {
+    return false;
+  } else {
+    for (let key in obj1) {
+      if (typeof obj1[key] === 'object' && typeof obj2[key] === 'object') {
+        return areObjectsEqual(obj1[key], obj2[key]);
```

Comment:
I believe this is not correct. If they are not equal you can return false. But if they are you need to keep iterating on all properties.

Reply:
this doesn't seem to be addressed.